packages/Search/examples/bvh_driver/bvh_driver.cpp

```diff
@@ -95,110 +46,97 @@ int main_( Teuchos::CommandLineProcessor &clp, int argc, char *argv[] )
+        Kokkos::View<int *, DeviceType> offset( "offset" );
+        Kokkos::View<int *, DeviceType> indices( "indices" );
+        start = std::chrono::high_resolution_clock::now();
+        bvh.query( queries, indices, offset );
+        end = std::chrono::high_resolution_clock::now();
+        elapsed_seconds = end - start;
+        os << "knn " << elapsed_seconds.count() << "\n";
     }
-    else if ( mode == "radius" )
-    {
-        Kokkos::View<double *, ExecutionSpace> radii( "radii", n_points );
-        auto radii_host = Kokkos::create_mirror_view( radii );
-
-        // use random radius for the search
-        // set the limit of approximately 100 points by
-        // solving n_points*pi*r^2/(Lx^2 + Ly^2 + Lz^2) <= 100
-        const int approx_points = 100;
-        double max_radius = sqrt(
-            approx_points * ( Lx * Lx + Ly * Ly + Lz * Lz ) / ( n * M_PI ) );
-        std::uniform_real_distribution<double> distribution_radius(
-            0.0, max_radius );
-        for ( int i = 0; i < n_points; ++i )
-        {
-            radii_host[i] = distribution_radius( generator );
-        }
-        Kokkos::deep_copy( radii, radii_host );
 
-        Kokkos::View<details::Within *, DeviceType> within_queries(
-            "within_queries", n_points );
+    {
+        Kokkos::View<DataTransferKit::Details::Within *, DeviceType> queries(
+            "queries", n_queries );
+        // radius chosen in order to control the number of results per query
+        // NOTE: minus "1+sqrt(3)/2 \approx 1.37" matches the size of the boxes
+        // inserted into the tree (mid-point between half-edge and
+        // half-diagonal)
+        double const pi = 3.14159265359;
+        double const r = 2. * std::cbrt( static_cast<double>( n_neighbors ) *
+                                         3. / ( 4. * pi ) ) -
+                         ( 1. + std::sqrt( 3. ) ) / 2.;
         Kokkos::parallel_for(
-            DTK_MARK_REGION( "register_within_queries" ),
-            Kokkos::RangePolicy<ExecutionSpace>( 0, n_points ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
             KOKKOS_LAMBDA( int i ) {
-                within_queries( i ) = details::within(
-                    {{point_coords( i, 0 ), point_coords( i, 1 ),
-                      point_coords( i, 2 )}},
-                    radii( i ) );
+                queries( i ) =
+                    DataTransferKit::Details::within( random_points( i ), r );
             } );
         Kokkos::fence();
 
-        Kokkos::View<int *, DeviceType> offset_within( "offset_within" );
-        Kokkos::View<int *, DeviceType> indices_within( "indices_within" );
-        bvh.query( within_queries, indices_within, offset_within );
+        Kokkos::View<int *, DeviceType> offset( "offset" );
+        Kokkos::View<int *, DeviceType> indices( "indices" );
+        start = std::chrono::high_resolution_clock::now();
```

Comment:
Can you add blank line before this one? I think it would be more readable to have a separate code block like
```c++
Kokkos::View<int *, DeviceType> offset( "offset" );
Kokkos::View<int *, DeviceType> indices( "indices" );

start = std::chrono::high_resolution_clock::now();
bvh.query( queries, indices, offset );
end = std::chrono::high_resolution_clock::now();

elapsed_seconds = end - start;
os << "radius " << elapsed_seconds.count() << "\n";
```